tests/src/python/test_qgsnetworkaccessmanager.py

```diff
@@ -92,7 +92,10 @@ def _preprocessor(request):
 
     def _on_reply_ready_read(self, reply):
         _bytes = reply.peek(reply.bytesAvailable())
-        self.assertEqual(_bytes.data().decode()[:14], "<!DOCTYPE html")
+        if isinstance(_bytes, bytes):
+            self.assertEqual(_bytes.decode()[:14], "<!DOCTYPE html")
+        else:
+            self.assertEqual(_bytes.data().decode()[:14], "<!DOCTYPE html")
```

Comment:
I've tested locally that below does the job both on QT5 and 6:
```suggestion
        self.assertEqual(bytes(_bytes).decode()[:14], "<!DOCTYPE html")
```